core/src/main/java/org/apache/accumulo/core/zookeeper/ZooSession.java

```diff
@@ -348,7 +377,13 @@ public ZooReaderWriter asReaderWriter() {
    * @return connection counter
    */
   public long getConnectionCounter() {
```

Comment:
Should this method also be marked as synchronized like the others?

Reply:
This method does not need to be synchronized as it reads the latest from a volatile.  Could possibly make it synchronized and further simplify the code, but that would require synchronizing a few other methods.  The zookeeper ref and zk counter need to be read and written as one. That is done by the way the current atomic ref is structured.  It could also be done by having two instance variables and always reading/writing either of those variables in a sync block.  Since the synchronization would cascade for making two instance variables will leave this as-is for now.